codalab/worker/docker_image_manager.py

```diff
@@ -99,33 +113,43 @@ def _cleanup(self):
         """
         while not self._stop:
```

Comment:
It seems that we have 2 `while not self._stop` here. One is within `_cleanup()` function, another is in `cleanup_loop()`. Do we have keep both of them? Let me know if I missed anything.

Reply:
Haha good catch! We don't need the 2 loops and this is likely a bug that may be causing issues. We should get rid of this one (in the `_cleanup` method)

Reply:
Okay, I removed the inner `while loop` and leave the outer `while loop`